Fix and test self-signed issuer

m-ildefons opened the pull request:
- Fix the name of the self-signed issuer in the annotation of the ingress.
- Add test ensuring the certificates and ingresses and annotations are created as expeced
- Fix broken tests for COSI RBAC resources

Fixes: aquarist-labs/s3gw#655

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] CHANGELOG.md has been updated should there be relevant changes in this PR.

m-ildefons committed:
Fix and test self-signed issuer

- Fix the name of the self-signed issuer in the annotation of the
  ingress.
- Add test ensuring the certificates and ingresses and annotations are
  created as expeced
- Fix broken tests for COSI RBAC resources

Fixes: aquarist-labs/s3gw#655
Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

PhanLe1010 commented on charts/s3gw/templates/ingress-traefik.yaml:
```diff
@@ -12,7 +12,7 @@ metadata:
     traefik.ingress.kubernetes.io/router.tls: "true"
     traefik.ingress.kubernetes.io/router.middlewares:
       '{{ .Release.Namespace }}-{{ include "s3gw.CORSMiddlewareName" . }}@kubernetescrd'
-    cert-manager.io/cluster-issuer: {{ default .Values.tlsIssuer .Values.customTlsIssuer | quote }}
```
This means that we don't support specifying customized tls issuer at all. I think we should allow the user to specify their own tls issuer

giubacc replied:
I agree with @PhanLe1010, it should be possible for the user to specify its tls issuer.

m-ildefons replied:
I fixed it. I was not aware this was a feature.

giubacc replied:
I dunno if we originally designed that to be an official feature, but we would break the existing if we changed that.

PhanLe1010 commented:
I tried this PR but it still failed when using `tlsIssuer: s3gw-letsencrypt-issuer`. I have created a new PR that fix this bug a some new bugs https://github.com/aquarist-labs/s3gw-charts/pull/119. Please let me know what you think @m-ildefons . Thank you

m-ildefons committed:
Add custom TLS issuer

Add an override option for a custom TLS issuer that is not the
letsencrypt issuer or the self-signed issuer.

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

m-ildefons committed:
custom tls issuer: fix private issuer

Fix cluster-internal private issuer if a custom TLS issuer is specified.

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

jecluis commented:
@m-ildefons what's the status on this one, and how closely related to #119 is this pull request?
